backend/tests/test_rate_limit_fix.py

```
@@ -0,0 +1,926 @@
+        # Verify limit_value is preserved
+        rate_limit = db_session.query(TenantRateLimit).filter(
+            TenantRateLimit.tenant_id == test_tenant.id,
+            TenantRateLimit.limit_type == "extractions_per_hour"
+        ).first()
+        
+        assert rate_limit.limit_value == configured_limit, "Limit value should be preserved"
+        assert rate_limit.current_usage == 2, "Usage should increment to 2"
+    
+    def test_check_rate_limit_compares_against_configured_value(
+        self,
+        db_session: Session,
```

Ruff (0.14.7)

Unused method argument: `db_session`

(ARG002)